Modify 'SameSite' cookie attribute to 'Strict'

neiltait opened the pull request:
This is intended to tighten up where the user's session cookie may be sent by the browser to improve security, particularly session-hijack and CSRF threats.

neiltait committed:
Modify 'SameSite' cookie attribute to 'Strict'

This is intended to tighten up where the user's session cookie may be sent by the browser to improve security, particularly session-hijack and CSRF threats.

pipwilson commented on server/middleware/setUpWebSession.ts:
```diff
@@ -21,7 +21,7 @@ export default function setUpWebSession(): Router {
       store,
       name: 'hmpps-sentence-plan-ui.session',
-      cookie: { secure: config.https, sameSite: 'lax', maxAge: config.session.expiryMinutes * 60 * 1000 },
+      cookie: { secure: config.https, sameSite: 'Strict', maxAge: config.session.expiryMinutes * 60 * 1000 },
```
this needs to be all-lowercase `strict`

pipwilson replied:
(see https://github.com/ministryofjustice/hmpps-sentence-plan-ui/actions/runs/13836930866/job/38714239108?pr=436#step:7:129)

neiltait committed:
Change 'S' to an 's' to get around TS checks

Typescript complains about the capitalisation of 'Strict', changing to lower case to keep it happy.

pipwilson committed:
Merge branch 'main' into samesite-strict

pipwilson reviewed (approved):
the code builds and the tests pass, what could possibly go wrong?
